lib/puppet/rest/route.rb

```diff
@@ -69,24 +72,8 @@ def with_base_url(dns_resolver)
       else
         # Otherwise, get server and port from default settings, taking
         # into account the server list for HA.
-        bound_server = Puppet.lookup(:server) do
-          if primary_server = Puppet.settings[:server_list][0]
-            primary_server[0]
-          else
-            Puppet.settings[:server]
-          end
-        end
-
-        bound_port = Puppet.lookup(:serverport) do
-          if primary_server = Puppet.settings[:server_list][0]
-            primary_server[1]
-          else
-            Puppet.settings[:masterport]
-          end
-        end
-
-        @server = bound_server
-        @port = bound_port
```

Comment:
Won't @default_server && @default_port always be present now that we have the :server and :masterport defaults?
So do we still need the else statement?

Reply:
You're right, I think we don't need any `if/else` here at all, it should be sufficient to just set `@server` and `@port` to the defaults here.